src/Umbraco.Core/Services/MediaService.cs

```diff
@@ -658,7 +677,7 @@ public void Move(IMedia media, int parentId, int userId = 0)
                 media.ParentId = parentId;
                 if (media.Trashed)
                 {
-                    media.ChangeTrashedState(false, parentId);
+                    ((Models.Media)media).ChangeTrashedState(false, parentId);

```

Comment:
How are we doing this for content? we need to remove any reason to cast like this otherwise it's not actually testable. Just had a look at the content service, seems this cleanup is part of http://issues.umbraco.org/issue/U4-4748 so no problem ;)